### PR TITLE
Add cli to support download/upload action

### LINKED ISSUE #18: Add cli for sdk to download model/dataset

### PR DESCRIPTION
**Pull request by SeanHH86**
fix #18

### CODE REVIEW

File: pycsghub/cli.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    "repoType": typer.Option(default=..., help="Specify the repository type."),
+    "repoID": typer.Option(default=..., help="Specify the repository ID."),
+    "repoFiles": typer.Option(default=..., help="Local path to the file or files to upload. Defaults to the relative path of the file of repo of OpenCSG Hub."),
+    "revision": typer.Option(default=..., help="An optional Git revision id which can be a branch name"),
+    "cache_dir": typer.Option(default=..., help="Path to the directory where to save the downloaded files."),
+    "endpoint": typer.Option(default=..., help="The address of the request to be sent."),
+    "token": typer.Option(default=..., help="A User Access Token generated from https://opencsg.com/settings/access-token"),
+}
+
+@app.command(name="download", help="Download model/dataset from OpenCSGS Hub")
```
Comment (Rader) [inline pycsghub/cli.py:19]: change `OpenCSGS Hub` to `opencsg.com`

File: setup.py [hunk 2]
```diff
@@ -6,12 +6,19 @@
...
-    packages=find_packages(),
     author="opencsg",
     author_email="contact@opencsg.com",
     long_description=long_description,
     long_description_content_type="text/markdown",
+    packages=find_packages(include="pycsghub*"),
+    include_package_data=True,
+    entry_points={
+        "console_scripts": [
+            "csghub=pycsghub.cli:app",
```
Comment (Rader) [inline setup.py:17]: command line name `csghub-cli` is better

File: pycsghub/cli.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import typer
+from typing import Annotated, List, Optional
+from pycsghub.cmd import repo
+from pycsghub.cmd.repo_types import RepoType
+from pycsghub.constants import DEFAULT_CSGHUB_DOMAIN, DEFAULT_REVISION
+
+app = typer.Typer(add_completion=False)
+
+OPTIONS = {
+    "repoType": typer.Option(default=..., help="Specify the repository type."),
```
Comment (Rader) [inline pycsghub/cli.py:10]: default to `model`

File: pycsghub/cli.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    "repoID": typer.Option(default=..., help="Specify the repository ID."),
+    "repoFiles": typer.Option(default=..., help="Local path to the file or files to upload. Defaults to the relative path of the file of repo of OpenCSG Hub."),
+    "revision": typer.Option(default=..., help="An optional Git revision id which can be a branch name"),
+    "cache_dir": typer.Option(default=..., help="Path to the directory where to save the downloaded files."),
+    "endpoint": typer.Option(default=..., help="The address of the request to be sent."),
+    "token": typer.Option(default=..., help="A User Access Token generated from https://opencsg.com/settings/access-token"),
```
Comment (Rader) [inline pycsghub/cli.py:16]: please support reading token from `env`

File: README_EN.md [hunk 5]
```diff
@@ -71,10 +71,51 @@ cd csghub-sdk
 pip install .
 ```
 
-## Use cases
+## Use cases of command line
+
+```shell
+# download model
+csghub download --repo-id=wanghh2000/myprivate1 --repo-type=model --token=3b77c98077b415ca381ded189b86d5df226e3776
```
Comment (Rader) [inline README_EN.md:78]: it's better to download a model like this

> csghub-cli download wanghh2000/myprivate1 

and download a dataset like this 

> csghub-cli download wanghh2000/myds1 --repo-type=dataset

**Review by Rader — CHANGES_REQUESTED** [review]
we can learn from [huggingface-cli](https://huggingface.co/docs/huggingface_hub/main/en/guides/cli)

File: pycsghub/cli.py (hunk 1 above)
Comment (SeanHH86) [reply pycsghub/cli.py]: fixed.

File: setup.py (hunk 2 above)
Comment (SeanHH86) [reply setup.py]: fixed.

File: pycsghub/cli.py (hunk 3 above)
Comment (SeanHH86) [reply pycsghub/cli.py:10]: it's already set at method #22.

repo_type: Annotated[RepoType, OPTIONS["repoType"]] = RepoType.MODEL,

File: pycsghub/cli.py (hunk 4 above)
Comment (SeanHH86) [reply pycsghub/cli.py:16]: those are optional, and get from env by default.

File: pycsghub/cli.py (hunk 3 above)
Comment (Rader) [reply pycsghub/cli.py:10]: > it's already set at method #22.
> 
> repo_type: Annotated[RepoType, OPTIONS["repoType"]] = RepoType.MODEL,

then, we can change the example to not input `--repo-type`

File: README_EN.md (hunk 5 above)
Comment (SeanHH86) [reply README_EN.md]: fixed.

**Review by Rader — APPROVED** [review]

**Pull request merged**